modules/oe_multilingual_url_suffix/src/Plugin/LanguageNegotiation/LanguageNegotiationUrlSuffix.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace Drupal\oe_multilingual_url_suffix\Plugin\LanguageNegotiation;
+
+use Drupal\Core\Language\LanguageInterface;
+use Drupal\Core\Render\BubbleableMetadata;
+use Drupal\language\Plugin\LanguageNegotiation\LanguageNegotiationUrl;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Class LanguageNegotiationUrlSuffix for identifying language via URL suffix.
+ *
+ * @LanguageNegotiation(
+ *   id = \Drupal\oe_multilingual_url_suffix\Plugin\LanguageNegotiation\LanguageNegotiationUrlSuffix::METHOD_ID,
+ *   types = {\Drupal\Core\Language\LanguageInterface::TYPE_INTERFACE,
+ *   \Drupal\Core\Language\LanguageInterface::TYPE_CONTENT,
+ *   \Drupal\Core\Language\LanguageInterface::TYPE_URL},
+ *   weight = -10,
+ *   name = @Translation("URL suffix"),
+ *   description = @Translation("Language from the URL (Path suffix)."),
+ *   config_route_name = "oe_multilingual_url_suffix.negotiation_url_suffix"
+ * )
+ */
+class LanguageNegotiationUrlSuffix extends LanguageNegotiationUrl {
+
+  /**
+   * The language negotiation method id.
+   */
+  const METHOD_ID = 'language-url-suffix';
+
+  /**
+   * The suffix delimiter.
+   */
+  const SUFFIX_DELIMITER = '_';
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getLangcode(Request $request = NULL) {
+    $langcode = NULL;
+
+    $config = $this->config->get('oe_multilingual_url_suffix.settings')->get('url_suffixes');
+    if ($request && $config) {
```

Comment:
In the base class method they check if the language manager is not NULL:
```php
    if ($request && $this->languageManager) {
      $languages = $this->languageManager->getLanguages();
```
Do we know why is that? Shouldn't we worry about it too? Language negotiator plugins are created in here:

```
\Drupal\language\LanguageNegotiator::getNegotiationMethodInstance()
```

```php
  public function getNegotiationMethodInstance($method_id) {
    if (!isset($this->methods[$method_id])) {
      $instance = $this->negotiatorManager->createInstance($method_id, []);
      $instance->setLanguageManager($this->languageManager);
      $instance->setConfig($this->configFactory);
      $instance->setCurrentUser($this->currentUser);
      $this->methods[$method_id] = $instance;
    }
    return $this->methods[$method_id];
  }
```
It seems not to be the plugin manager that injects dependencies.

Reply:
Unfortunately can't find reason and a specially why this is not done through dependency injections.
I have updated our implementation. of this method. Yes. Some cases was missed. This method could return langcode of language which is removed already.